libs/ui/src/print_element.tsx

```diff
@@ -12,18 +12,45 @@ const ScreenDisplayNone = styled.div`
   }
 `;
 
-export interface PrintElementOptions extends PrintOptions {
+interface PrintElementBaseOptions {
   screenDisplayNone?: boolean;
 }
 
+export type PrintElementToPdfOptions = PrintElementBaseOptions;
+
+export interface PrintElementOptions
+  extends PrintElementBaseOptions,
+    PrintOptions {}
+
+type PrintElementToPdfOptionsDiscriminated = PrintElementToPdfOptions & {
+  printToPdf: true;
+};
+type PrintElementOptionsDiscriminated = PrintElementOptions & {
+  printToPdf: false;
+};
+type PrintElementSuperOptionsDiscriminated =
+  | PrintElementToPdfOptionsDiscriminated
+  | PrintElementOptionsDiscriminated;
+
 // Render an element and print it. The function to render the element takes a
 // callback to indicate when the component has finished rendering and is ready
 // to be printed. This accommodates components that may want to do multiple
 // renders or post-processing before being ready to print.
-export async function printElementWhenReady(
+async function printElementWhenReadySuper(
   elementWithOnReadyCallback: ElementWithCallback,
-  { screenDisplayNone = true, ...printOptions }: PrintElementOptions
-): Promise<void> {
+  printElementToPrinterOptions: PrintElementOptionsDiscriminated
+): Promise<void>;
+async function printElementWhenReadySuper(
+  elementWithOnReadyCallback: ElementWithCallback,
+  printElementToPdfOptions: PrintElementToPdfOptionsDiscriminated
+): Promise<Uint8Array>;
+async function printElementWhenReadySuper(
+  elementWithOnReadyCallback: ElementWithCallback,
+  {
+    screenDisplayNone = true,
+    ...printOptionsDiscriminated
+  }: PrintElementSuperOptionsDiscriminated
+): Promise<void | Uint8Array> {
```

Comment:
@eventualbuddha, I'm not explicitly looking for your review on this one, but as the person who explained overloads to me I'm curious about your thoughts on the overloading in this file, use of discriminators, etc.

Reply:
Helper function is a great call, thank you.